src/MongoDB.Driver.Core/Core/Connections/IsMasterResult.cs

```diff
@@ -322,6 +333,22 @@ public ServerType ServerType
             }
         }
 
+        /// <summary>
+        /// Get the SpeculativeAuthenticate reply.
+        /// </summary>
+        /// <value>
+        /// Null if isMaster["ok"] !=1 or if the SpeculativeAuthenticate reply was not included in the isMaster response.
+        /// </value>
+        public BsonDocument SpeculativeAuthenticate
+        {
+            get
+            {
+                return _wrapped.GetValue("ok", defaultValue: false).ToBoolean()
+                    ? (BsonDocument)_wrapped.GetValue("speculativeAuthenticate", defaultValue: null)
+                    : null;
```

Comment:
Thanks Robert! My original comment appears to have gotten lost, but I've taken your refactor to avoid using default values, but wrapped it in a ternary instead :) If you feel strongly that the non-ternary version is easier to read, I can revert it to your version.

Reply:
You know how I feel about ternaries... specially ones that don't fit on one line.

Multi line ternaries are also difficult to format (and to read). When an expression is more than one line long we like all lines except the last to end with an operator (which visually indicates that there is more to come), but the way you've been formatting ternaries violates that.

I think we should avoid complex ternaries that don't fit on one line.

Reply:
nit: our standard formatting with this multi-line ternary would be:

    return 
        _wrapped.TryGetValue("ok", out var ok) && ok.ToBoolean() &&
        _wrapped.TryGetValue("speculativeAuthenticate", out var speculativeAuthenticate) ?
            (BsonDocument)speculativeAuthenticate :
            null;

but better yet just avoid multi-line ternaries.